DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/Rules/AnonymizerRuleFactory.cs

```diff
@@ -0,0 +1,136 @@
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Dicom;
+using EnsureThat;
+using Microsoft.Health.Dicom.Anonymizer.Core.AnonymizerConfigurations;
+using Microsoft.Health.Dicom.Anonymizer.Core.Exceptions;
+using Microsoft.Health.Dicom.Anonymizer.Core.Model;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Dicom.Anonymizer.Core.Rules
+{
+    public class AnonymizerRuleFactory : IAnonymizerRuleFactory
+    {
+        private readonly AnonymizerDefaultSettings _defaultSettings;
+
+        private readonly Dictionary<string, JObject> _customizedSettings;
+
+        private readonly IAnonymizerProcessorFactory _processorFactory;
+
+        private readonly IAnonymizerSettingsFactory _settingsFactory;
+
+        public AnonymizerRuleFactory(AnonymizerConfiguration configuration, IAnonymizerProcessorFactory processorFactory = null, IAnonymizerSettingsFactory settingsFactory = null)
+        {
+            EnsureArg.IsNotNull(configuration, nameof(configuration));
+
+            _defaultSettings = configuration.DefaultSettings;
+            _customizedSettings = configuration.CustomizedSettings;
+            _processorFactory = processorFactory;
+            _settingsFactory = settingsFactory;
+        }
+
+        public AnonymizerRule[] CreateAnonymizationDicomRule(JObject[] ruleContent)
+        {
+            return ruleContent?.Select(entry => CreateAnonymizationDicomRule(entry)).ToArray();
+        }
+
+        public AnonymizerRule CreateAnonymizationDicomRule(JObject ruleContent)
+        {
+            EnsureArg.IsNotNull(ruleContent, nameof(ruleContent));
+
+            // Parse and validate method
+            if (!ruleContent.ContainsKey(Constants.MethodKey))
+            {
+                throw new AnonymizationConfigurationException(DicomAnonymizationErrorCode.MissingConfigurationFields, "Missing method in rule config");
+            }
+
+            var method = ruleContent[Constants.MethodKey].ToString();
+            var supportedMethods = Enum.GetNames(typeof(AnonymizerMethod)).ToHashSet(StringComparer.InvariantCultureIgnoreCase);
+            if (!supportedMethods.Contains(method))
+            {
+                throw new AnonymizationConfigurationException(DicomAnonymizationErrorCode.UnsupportedAnonymizationRule, $"Anonymization method {method} not supported.");
+            }
+
+            // Parse and validate settings
+            JObject parameters = null;
+            if (ruleContent.ContainsKey(Constants.Parameters))
+            {
+                parameters = ruleContent[Constants.Parameters].ToObject<JObject>();
+            }
+
+            JObject ruleSetting = _defaultSettings.GetDefaultSetting(method);
+            if (ruleContent.ContainsKey(Constants.RuleSetting))
```

Comment:
>ruleContent.ContainsKey(Constants.RuleSetting) [](http://example.com/codeflow?start=16&length=46)

Don't quite understand the logic here.
There's no example of "setting" in configuration json file.
Neither in the unit tests.

Reply:
And looks like for each rule, users can specify exactly one custom setting?

Reply:
![image](https://user-images.githubusercontent.com/57981365/121293511-52c86a00-c91e-11eb-84df-5df6c6328f75.png)

Reply:
Since the changes of functions, some unit tests are removed.
More tests will be added later.